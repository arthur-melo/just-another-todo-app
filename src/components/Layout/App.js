import React from 'react';

import './App.css';

import Header from '../Header/Header';
import ContentContainer from '../../containers/ContentContainer';
import Footer from '../Footer/Footer';

const App = () => (
  <div className="app__container">
    <Header className="app__header" />
    <ContentContainer className="app__content" />
    <Footer className="app__footer" />
  </div>
);

export default App;
