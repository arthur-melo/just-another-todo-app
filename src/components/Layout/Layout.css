.layout__header {
  min-height: 10vh;
}

.layout__footer {
  min-height: 10vh;
}
