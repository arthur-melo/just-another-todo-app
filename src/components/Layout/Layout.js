import PropTypes from 'prop-types';

import './Layout.css';

import Header from './Header/Header';
import Footer from './Footer/Footer';

const layoutPropTypes = {
  children: PropTypes.node.isRequired,
};

const Layout = props => {
  const { children } = props;

  return (
    <div className="d-flex flex-column w-100 h-100">
      <header className="layout__header flex-shrink-0 w-100">
        <Header />
      </header>
      <main className="flex-grow-1 flex-shrink-0 mt-1">{children}</main>
      <footer className="layout__footer flex-shrink-0 w-100">
        <Footer />
      </footer>
    </div>
  );
};

Layout.propTypes = layoutPropTypes;

export default Layout;
