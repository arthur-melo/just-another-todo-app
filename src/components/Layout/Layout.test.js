import renderer from 'react-test-renderer';

import Layout from './Layout';

describe('Layout', () => {
  it('should render the Layout component', () => {
    const child = <div>Children</div>;
    const component = <Layout>{child}</Layout>;

    const view = renderer.create(component).toJSON();
    expect(view).toMatchSnapshot();
  });
});
