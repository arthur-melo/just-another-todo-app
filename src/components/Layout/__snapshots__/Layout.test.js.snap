// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render the Layout component 1`] = `
<div
  className="layout__container"
>
  <header
    className="layout__header"
  >
    <div
      className="header__container"
    >
      <nav
        className="navbar"
      >
        <span
          className="header__span navbar-brand"
        >
          <svg
            alt="List emoji"
            aria-hidden="true"
            className="svg-inline--fa fa-rectangle-list fa-lg d-inline-block"
            data-icon="rectangle-list"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 96C0 60.7 28.7 32 64 32H512c35.3 0 64 28.7 64 64V416c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V96zM128 288c17.7 0 32-14.3 32-32s-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32zm32-128c0-17.7-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32s32-14.3 32-32zM128 384c17.7 0 32-14.3 32-32s-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32zm96-248c-13.3 0-24 10.7-24 24s10.7 24 24 24H448c13.3 0 24-10.7 24-24s-10.7-24-24-24H224zm0 96c-13.3 0-24 10.7-24 24s10.7 24 24 24H448c13.3 0 24-10.7 24-24s-10.7-24-24-24H224zm0 96c-13.3 0-24 10.7-24 24s10.7 24 24 24H448c13.3 0 24-10.7 24-24s-10.7-24-24-24H224z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
        <span
          className="header__span"
        >
          Just Another Todo App
        </span>
      </nav>
    </div>
  </header>
  <main
    className="layout__content"
  >
    <div>
      Children
    </div>
  </main>
  <footer
    className="layout__footer"
  >
    <div
      className="footer__container text-center d-flex h-100 justify-content-center align-items-center"
    >
      <div
        className="wrapper"
      >
         
        <p
          className="footer__p"
        >
          This app runs offline and doesn't use analytics.
        </p>
        <p
          className="footer__p"
        >
          Keep your data safe in your browser!
        </p>
        <p
          className="footer__p"
        >
          Made with 
           by
           
          <a
            href="https://github.com/arthur-melo"
            rel="noopener noreferrer"
            target="_blank"
          >
            @arthur-melo
          </a>
        </p>
      </div>
    </div>
  </footer>
</div>
`;
